app/models/text_field_type.rb

```diff
@@ -25,14 +28,23 @@ def mapping_field_name
   end
 
   def text_present
-    errors.add(:text, "must be present") if @text.empty?
+    errors.add(:text, 'must be present') if @text.empty?
   end
 
   def text_length
     validator = LengthValidator.new(validations[:length].merge(attributes: [:text]))
     validator.validate_each(self, :text, text)
   end
 
+  def text_unique
+    errors.add(:base, 'Slug Must be unique') unless FieldItem.jsonb_contains(:data, text: text).empty?
+    true
+  end
+
+  def validate_uniqueness?
```

Comment:
That's what I was referring to - the extraction comment was made under the belief that had already happened. Still I maintain that the validation itself should be presented by ActiveModel, like we're doing with presence